### PR TITLE
Server route termbyid

### PR DESCRIPTION
**Pull request by karishma-gangwani**
## Description
@xzhou82 I think term1_q was missing from the request. All tests pass on my local now. Let me know. 

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/termdb/TermdbVocab.js [hunk 1]
```diff
@@ -527,7 +535,7 @@ export class TermdbVocab extends Vocab {
 			if (filter) {
 				body.filter = getNormalRoot(filter)
 			}
-			return await dofetch3('/termdb', { body })
+			return await dofetch3('termdb/categories', { headers, body })
```
Comment (xzhou82) [inline client/termdb/TermdbVocab.js:538]: this breaks [snplst term](http://localhost:3000/?noheader=1&mass={%22genome%22:%22hg38%22,%22dslabel%22:%22SJLife%22,%22nav%22:{%22activeTab%22:1},%22plots%22:[{%22chartType%22:%22regression%22,%22regressionType%22:%22linear%22,%22outcome%22:{%22id%22:%22LV_Ejection_Fraction_3D%22},%22independent%22:[{%22term%22:{%22type%22:%22snplst%22,%22snps%22:[{%22rsid%22:%22rs1333261826%22,%22effectAllele%22:%22T%22},{%22rsid%22:%22rs1010330999%22,%22effectAllele%22:%22G%22},{%22rsid%22:%22rs1641548%22,%22effectAllele%22:%22T%22},{%22rsid%22:%22rs858528%22,%22effectAllele%22:%22T%22},{%22rsid%22:%22rs1642793%22,%22effectAllele%22:%22C%22},{%22rsid%22:%22rs1042522%22},{%22rsid%22:%22rs1642782%22},{%22rsid%22:%22rs6503048%22},{%22rsid%22:%22rs1472752888%22,%22effectAllele%22:%22T%22}]},%22q%22:{%22AFcutoff%22:5,%22alleleType%22:0,%22geneticModel%22:0,%22missingGenotype%22:0,%22restrictAncestry%22:{%22name%22:%22European%20ancestry%22,%22tvs%22:{%22term%22:{%22id%22:%22genetic_race%22,%22type%22:%22categorical%22,%22name%22:%22Genetically%20defined%20race%22},%22values%22:[{%22key%22:%22European%20Ancestry%22,%22label%22:%22European%20Ancestry%22}]}}}},{%22id%22:%22sex%22,%22refGrp%22:%222%22}]}]}), please revert. see my previous commit

**Review by xzhou82 — APPROVED** [review]

File: client/common/termutils.js [hunk 2]
```diff
@@ -22,7 +22,13 @@ export async function getterm(termid, dslabel = null, genome = null) {
 	}
 	if (!dslabel) throw 'getterm: dslabel missing'
 	if (!genome) throw 'getterm: genome missing'
-	const data = await dofetch3(`termdb?dslabel=${dslabel}&genome=${genome}&gettermbyid=${termid}`)
+	const body = {
```
Comment (xzhou82) [inline client/common/termutils.js:25]: i don't see `getterm()` exported from here used anywhere. it duplicates vocab api. can try to delete this function and test

**Comment by xzhou82** [comment]
per your question, q{} is only required for termsetting. getting a term object won't need it

File: client/termdb/TermdbVocab.js (hunk 1 above)
Comment (karishma-gangwani) [reply client/termdb/TermdbVocab.js:538]: okay. Let me see. For now, I have converted it to a draft.

**Comment by xzhou82** [comment]
thanks. this must have been created early on before vocab

**Comment by siosonel** [comment]
@xzhou82 and @karishma-gangwani I removed the `getterms()` from `termutils.js`, and fixed the failing vocab tests. Please see if the integrations tests pass for you locally. If so, we can convert the PR to ready.

**Comment by karishma-gangwani** [comment]
all tests pass! thank you both! we can merge

**Review by siosonel — APPROVED** [review]

**Comment by siosonel** [comment]
looks good to me!

File: server/routes/termdb.termbyid.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	}
+}
+
+function init({ genomes }) {
+	return async (req: any, res: any): Promise<void> => {
+		const q = req.query // as getcategoriesRequest
+		try {
+			const g = genomes[req.query.genome]
+			const ds = g.datasets[req.query.dslabel]
+			const tdb = ds.cohort.termdb
```
Comment (xzhou82) [inline server/routes/termdb.termbyid.ts:44]: @karishma-gangwani just to point out this logical err. if ds is undefined, this line crashes server

Comment (siosonel) [reply server/routes/termdb.termbyid.ts:44]: @xzhou82  I can fix this in my branch

Comment (karishma-gangwani) [reply server/routes/termdb.termbyid.ts:44]: okay. thanks for pointing it out. were you able to fix it directly in master?

**Comment by xzhou82** [comment]
please don't. i've fixed it in my PR

**Pull request merged**